nylas/client/uas_models.py

```diff
@@ -0,0 +1,330 @@
+        if grant_id:
+            request["grant_id"] = grant_id
+        if login_hint:
+            request["login_hint"] = login_hint
+        if state:
+            request["state"] = state
+        if expires_in:
+            request["expires_in"] = expires_in
+        if settings:
+            request["settings"] = settings
+        if metadata:
+            request["metadata"] = metadata
+        if scope:
+            request["scope"] = scope
+
+        response = self.api._post_resource(Grant, "auth", None, request, path="connect")
+        if "data" in response:
+            response = response["data"]
+
+        return response
+
+    def _set_integrations_api_url(self):
+        self.api.api_server = "https://{app_name}.{region}.nylas.com".format(
+            app_name=self.app_name, region=self.region.value
+        )
+
+    class Region(str, Enum):
+        """
+        This is an Enum the regions supported by the Integrations API
+        """
+
+        US = "us"
+        EU = "eu"
+
+    class Provider(str, Enum):
+        """
+        This is an Enum representing all the available providers for integrations
+        """
+
+        GOOGLE = "google"
+        MICROSOFT = "microsoft"
+        IMAP = "imap"
+        ZOOM = "zoom"
+
+
+class UASRestfulModelCollection(RestfulModelCollection):
+    def __init__(self, model_class, api):
+        RestfulModelCollection.__init__(self, model_class, api)
+
+    def _get_model_collection(self, offset=0, limit=CHUNK_SIZE):
+        filters = copy(self.filters)
+        filters["offset"] = offset
+        if not filters.get("limit"):
+            filters["limit"] = limit
+
+        response = self.api._get_resource_raw(self.model_class, None, **filters).json()
+        if "data" not in response or response["data"] is None:
+            return []
+
+        return [
+            self.model_class.create(self, **x)
+            for x in response["data"]
+            if x is not None
+        ]
+
+
+class IntegrationRestfulModelCollection(UASRestfulModelCollection):
+    def __init__(self, api):
+        RestfulModelCollection.__init__(self, Integration, api)
+
+    def get(self, provider):
+        """
+        Get an existing integration for a provider
+
+        Args:
+            provider (UAS.Provider): The provider
+
+        Returns:
+            Integration: The existing integration
+        """
+        return super(IntegrationRestfulModelCollection, self).get(provider.value)
+
+    def delete(self, provider, data=None, **kwargs):
+        """
+        Deletes an existing integration for a provider
+
+        Args:
+            provider (UAS.Provider): The provider
+        """
+        super(IntegrationRestfulModelCollection, self).delete(
+            provider.value, data=data, **kwargs
+        )
+
+
+class GrantRestfulModelCollection(UASRestfulModelCollection):
+    def __init__(self, api):
+        RestfulModelCollection.__init__(self, Grant, api)
+
```

Comment:
this should also have a get and delete just like integration 🤔

Reply:
@pengfeiye `RestfulModelCollection` supports all the typical methods like get, delete, create, etc. but for Integration I had to override it because, unlike Grant, it doesn't have an ID field -- but a provider field. So all it does is insures that the user passes in a Provider instead of an ID.